webapp/channels/src/components/more_channels/more_channels.tsx

```diff
@@ -92,13 +95,24 @@ export default class MoreChannels extends React.PureComponent<Props, State> {
     }
 
     componentDidMount() {
-        this.props.actions.getChannels(this.props.teamId, 0, CHANNELS_CHUNK_SIZE * 2);
+        this.props.actions.getChannels(this.props.teamId, 0, CHANNELS_CHUNK_SIZE * 2).then((result) => {
+            this.props.actions.getChannelsMemberCount(result.data?.map((channel) => channel.id) || []);
+        });
         if (this.props.canShowArchivedChannels) {
-            this.props.actions.getArchivedChannels(this.props.teamId, 0, CHANNELS_CHUNK_SIZE * 2);
+            this.props.actions.getArchivedChannels(this.props.teamId, 0, CHANNELS_CHUNK_SIZE * 2).then((result) => {
+                this.props.actions.getChannelsMemberCount(result.data?.map((channel) => channel.id) || []);
+            });
         }
         this.loadComplete();
     }
 
+    componentDidUpdate(prevProps: Props) {
+        if (this.props.channels.length !== prevProps.channels.length || this.props.archivedChannels.length !== prevProps.archivedChannels.length) {
+            const allChannelIds = this.props.channels.map((channel) => channel.id).concat(this.props.archivedChannels.map((channel) => channel.id));
+            this.props.actions.getChannelsMemberCount(allChannelIds);
```

Comment:
Since we're calling `getChannelsMemberCount` on mount and on `nextPage`, I don't think we also need it in `componentDidUpdate` any more

Reply:
yeah, in the beginning, I was missing calling on `nextPage` so I thought this was necessary